test/sqllogictest/chbench.slt

```diff
@@ -1138,16 +1137,16 @@ ORDER BY supplier_cnt DESC
 | | implementation = DeltaQuery
 | |   delta %0 %1.(#0)
 | |   delta %1 %0.(#0)
-| | demand = (#17, #20..#22)
```

Comment:
Nit, but: this PR removes the `demand` printing but doesn't remove the `Demand` transform, afaiu. Which still remains important because it is the only inter-view, intra-dataflow way to push the same information around.

Reply:
I did not remove the `demand` printing. It's just that:
* `JoinImplementation` changes the `demand` field to `None`
* I removed the call to the `Demand` transform that runs after `JoinImplementation`, so the `demand` field is always `None`.
* `demand` is not printed when it is equal to `None`.

After landing projection pushdown, I can follow up by replacing `Demand` with `ProjectionPushdown` for inter-view pushing of projection information.

Reply:
Reasons for removing the call to the `Demand` transform that runs after `JoinImplementation`:
* Making the `demand` field of a Join to be not `None` is no longer necessary because projection pushdown has put a project around the join that will be fused at the LIR stage.
* Propagating the `col1 = col2` information should have already been done in the physical transformation step.